### DISCUSSION TITLE
Possible to return multiple fixing objects, each of which call `replaceTextRange` as long as the ranges do not overlap?

### DISCUSSION THREAD

**Discussion by sstchur** [eslint/eslint#16358] [Q&A]
The title says it all, but I can't get it to work.
I have a full github repository that repros the issue I'm facing, but briefly:

```typescript
fix(fixer) {
   return [
      fixer.replaceTextRange([9,13], '// C'),
      fixer.replaceTextRange([22,26], '// B')
   ]
}
```

I'm returning two fixing objects.  They uses ranges that do not overlap.  The first one works. It replaces the original code: `// A` with `// C`, but the second does not.  It should replace `// B` with `// D`.

Am I doing something wrong?  I read all the docs, and by my interpretation, this ought to work.

**Comment by mdjermanovic** [comment]
> ```ts
> fix(fixer) {
>    return [
>       fixer.replaceTextRange([9,13], '// C'),
>       fixer.replaceTextRange([22,26], '// B')
>    ]
> }
> ```
> but the second does not. It should replace `// B` with `// D`.

Returning an array of fixing objects should work. In this example, the problem could be a typo - if you want to replace with `// D`, then it should be `fixer.replaceTextRange([22,26], '// D')`.

**Reply by sstchur** [reply]
Good catch! Sorry about the typo, but even once that is fixed, I can still see the issue.  I've updated my github repository with a more complete example: https://github.com/sstchur/eslint-repro

In this example, I have (obviously contrived, but that's not the point) code, like:

```typescript
// A
// B
class Person {
  constructor(A: string, B: string) {}
}
```

And the ESLint rule looks for Identifiers and returns two fixing objects.  In the case of A, the rule returns fixing objects with non-overlapping, hardcoded ranges that correspond to `// A` and `A`, and in the case of B, it returns two fixing objects that correspond to the ranges of `// B` and `B`.

In the case of A, everything seems to work, but in the case of B, it doesn't get applied.  This feels strange.  If anything, I would expect the fixer for B (assuming it runs second) to overwrite the changes from the fixer for A (although, my expectation is that the fixers are applied independently and do not impact one another).

Here's the output I get after both fixers have run:

```
Difference:

- Expected
+ Received

          // C
-         // D
+         // B
          class Person {
-           constructor(C: string, D: string) {}
+           constructor(C: string, B: string) {}
          }
```

We can see that `// A` and `A` got changed to `// C` and `C`, but `// B` and `B` did NOT get changed to `// D` and `D`. 🤔

Any ideas what I'm still doing wrong?
